iothub/device/src/Authentication/IotHubConnectionCredentials.cs

```diff
@@ -405,24 +405,24 @@ private static IAuthenticationMethod GetAuthenticationMethodFromConnectionString
             else if (iotHubConnectionString.SharedAccessKey != null)
             {
                 return iotHubConnectionString.ModuleId == null
-                    ? new DeviceAuthenticationWithRegistrySymmetricKey(
+                    ? new ClientAuthenticationWithRegistrySymmetricKey(
+                        iotHubConnectionString.SharedAccessKey,
+                        iotHubConnectionString.DeviceId)
+                    : new ClientAuthenticationWithRegistrySymmetricKey(
```

Comment:
Are there 2 different ctors? If so, why not have just one with an optional module Id parameter, which you can pass in anyway. If it is null, no problem.

Reply:
Ah, they are using the same ctor. I missed this part, will change it.